### PR TITLE
CVP-1612 updated flow for address field types to use addressQueryBuilder

### PR DESCRIPTION
**Pull request by mintusah25**
Changes:

1. string.js - Updated filtering logic to use addressQueryBuilder() for searching on Address.
2. Created test cases specifically for search_by_address.

### CODE REVIEW

File: src/tests/searchParameters/searchByAddress/searchByAddress.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                .set(getHeaders());
+            // noinspection JSUnresolvedFunction
+            expect(resp).toHaveMergeResponse({created: true});
+
+            // search by token system and code and make sure we get the right Medication back
+            resp = await request
+                .get('/4_0_0/Person/?address=Berea&_bundle=1')
+                .set(getHeaders());
+            // noinspection JSUnresolvedFunction
+            expect(resp).toHaveResponse(expectedPersonResources2);
```
Comment (anmolmittal1) [inline src/tests/searchParameters/searchByAddress/searchByAddress.test.js:92]: 1. Add test case for invalid search
2. Add test for postal code as well

File: src/tests/searchParameters/searchByAddress/fixtures/expected/expectedSearchByAddressCity.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            }
+          ]
+        }
+      }
+    ],
+    "resourceType": "Bundle",
+    "total": 0,
+    "type": "searchset"
+  }
+
```
Comment (anmolmittal1) [inline src/tests/searchParameters/searchByAddress/fixtures/expected/expectedSearchByAddressCity.json:257]: Fix this invalid EOF

File: src/tests/searchParameters/searchByAddress/fixtures/Person/person1.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "resourceType": "Person",
+    "id": "1",
+    "meta": {
+        "versionId": "1",
+        "lastUpdated": "2022-06-13T13:52:25.000Z",
+        "security": [
+            {
+                "system": "https://www.icanbwell.com/access",
+                "code": "unitypoint"
+            },
```
Comment (anmolmittal1) [inline src/tests/searchParameters/searchByAddress/fixtures/Person/person1.json:11]: 1. format all json files with 2 space indentation
2. Can we create a single bundle for all the patients?

**Review by anmolmittal1 — CHANGES_REQUESTED** [review]

File: src/operations/query/filters/string.js [hunk 4]
```diff
@@ -56,6 +56,13 @@ function filterByString({queryParameterValue, propertyObj, columns}) {
         return andSegments;
     }
 
+    // If he field is adress, use address query builder to apply the search in all address attributes
```
Comment (007sunny) [inline src/operations/query/filters/string.js:59]: Typos

**Review by 007sunny — APPROVED** [review]

**Review by anmolmittal1 — APPROVED** [review]

**Pull request merged**